### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-navigation-item

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1481]
## Description
This task describes the Storybook implementation for the sd-navigation-item component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-navigation-item can be found here: [sd-navigation-item Notes](https://www.notion.so/matuzo/sd-navigation-item-dc889e6da8ed49d3932deba051ffe20e).

-------------------------

## Comment
**Problem:** Accordion items don't provide enough semantic information.
**Solution:** Add and toggle aria-expanded 

```
<button aria-expanded="false" aria-controls="navigation-item-details">
    Navigation
</button>
<div id="navigation-item-details" hidden>
    …
</div>
```

**Problem:** aria-controls on accordion items points to a non-existent id ("navigation-item-details").
**Solution:** Fix the relation or remove the attribute.

**Problem:** The accessible name of the button in the “separated” example is "chevron down button".
**Solution:** Provide a label that describes the button's purpose.

## Notes
- Disabled navigation items: See [@sd-button](https://www.notion.so/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=24). Everything I wrote there about disabling form elements applies here as well.
- Labels of buttons and links should be descriptive but concise. Items with descriptions create long labels.

## Best Practices
- Use links for navigation and buttons for performing other actions 

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
